test/e2e/node: add [NodeConformance] label to ConfigMap update test

rbiamru opened the pull request:
#### What type of PR is this?

/kind cleanup
/sig node
/area conformance
/priority backlog

#### What this PR does / why we need it:

This PR picks up a small part of the work from the help-wanted issue kubernetes/test-infra#33827. It updates the ConfigMap update test in `test/e2e/common/node/configmap.go` to use `framework.ConformanceIt` with `f.WithNodeConformance()`.

- Ensures the test is properly marked as [NodeConformance].
- Removes it from the unlabeled test lane in TestGrid.
<img width="1168" height="592" alt="Screenshot 2025-09-15 001232" src="https://github.com/user-attachments/assets/65117003-5942-4b6a-8c44-9059726cf285" />
- Brings it in line with other ConfigMap tests in the same file, which are already part of the conformance suite.

#### Which issue(s) this PR is related to:

Ref: kubernetes/test-infra#33827


#### Special notes for your reviewer:

This is a small, incremental step towards cleaning up unlabeled tests so we can eventually drop the `*-unlabelled` TestGrid lane.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

rbiamru committed:
test/e2e/node: add [NodeConformance] label to ConfigMap update test

rbiamru commented:
/assign @SergeyKanzhelev

This PR updates the ConfigMap update test in `test/e2e/common/node/configmap.go` to use `ConformanceIt` with `f.WithNodeConformance()`, aligning it with other ConfigMap tests and removing it from the unlabeled lane (#33827). Thanks!

SergeyKanzhelev commented on test/e2e/common/node/configmap.go:
```diff
@@ -141,7 +141,7 @@ var _ = SIGDescribe("ConfigMap", func() {
 	})
 
-	ginkgo.It("should update ConfigMap successfully", func(ctx context.Context) {
+	framework.ConformanceIt("should update ConfigMap successfully", f.WithNodeConformance(), func(ctx context.Context) {
```
wouldn't the `ConformanceIt` require the special comment (with the release and such) and an update of conformance.yaml file?

dims replied:
@SergeyKanzhelev the verify scripts should fail and indicate that - `require the special comment (with the release and such) and an update of conformance.yaml file?`

rbiamru committed:
test/e2e/node: promote ConfigMap update test to Conformance

rbiamru committed:
fix gofmt

rbiamru commented:
Added the conformance block, updated conformance.yaml, verified with ./hack/verify-conformance-yaml.sh and ./hack/verify-gofmt.sh. 
Checked git blame (#80787, #114401, #112923) — no indication this test was intentionally excluded from conformance. 
Also no flaky history in TestGrid/issues. 
@SergeyKanzhelev

SergeyKanzhelev reviewed (approved):
/lgtm
/approve

dims commented:
@rbiamru @SergeyKanzhelev we'll need evidence for `it is stable and runs consistently (e.g., no flakes), and has been running for at least two weeks` from the check list here - https://github.com/kubernetes/community/blob/83f1975f3ebdaffeb8dbc5aaeea377725ab0342a/contributors/devel/sig-architecture/conformance-tests.md

Can you confirm that the failures in our triage board are not related to the test itself?
https://storage.googleapis.com/k8s-triage/index.html?test=should%20update%20ConfigMap%20successfully
